Sources/EventProducer/Common/Scheduler.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+protocol Scheduler: AnyObject {
+	var schedulingTime: TimeInterval { get }
+	func timerTriggered(headerHelper: HeaderHelper) async throws
+	var schedulerTask: Task<Void, Error>? { get set }
+}
+
+extension Scheduler {
+	/// The Scheduler applies a best-effort approach to sending events to the TL Consumer, meaning that it continuously tries to send
+	/// events while the app is executing,
+	/// regardless of current internet conditions or if the app considers itself to be online or not. As long as the app is
+	/// executing, the Scheduler shall be running
+	func runScheduling(with headerHelper: HeaderHelper) {
+		guard schedulerTask == nil else {
+			return
+		}
+		
+		schedulerTask = Task { [weak self, headerHelper] in
+			repeat {
+				guard let self else {
+					return
+				}
+				
+				let schedulingTimeInNanoseconds = UInt64(schedulingTime) * NSEC_PER_SEC
+				
+				try await Task.sleep(nanoseconds: schedulingTimeInNanoseconds)
```

Comment:
In player we have a convenience method in a extension for sleeping in seconds
```
static func sleep(seconds: Double) async throws
```

Maybe we can move it to Common.. but can be done in a cleanup Pr later.

Reply:
great idea! I think it is worth to do it in a cleanup PR